MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -417,45 +417,58 @@ namespace WPEFramework {
-                    if (joGetResult.HasLabel("success") && joGetResult["success"].Boolean()) {
-                        static const char* kDeviceInitializationContext = "deviceInitializationContext";
-                        if (joGetResult.HasLabel(kDeviceInitializationContext)) {
-                            LOGINFO("%s found in the response", kDeviceInitializationContext);
-                            success = setDeviceInitializationContext(joGetResult);
+            do 
+            {
+                if ((getServiceState(m_service, secMgr_callsign, state) == Core::ERROR_NONE) && (state == PluginHost::IShell::state::ACTIVATED)) {
+                    LOGINFO("%s is active", secMgr_callsign);
+
+                    thunder_client=getThunderPluginHandle(secMgr_callsign_ver);
+                    if (thunder_client != nullptr) {
+                        JsonObject params;
+                        JsonObject joGetResult;
+
+                        thunder_client->Invoke<JsonObject, JsonObject>(5000, "getDeviceInitializationContext", params, joGetResult);
+                        if (joGetResult.HasLabel("success") && joGetResult["success"].Boolean()) {
+                            static const char* kDeviceInitializationContext = "deviceInitializationContext";
+                            if (joGetResult.HasLabel(kDeviceInitializationContext)) {
+                                LOGINFO("%s found in the response", kDeviceInitializationContext);
+                                success = setDeviceInitializationContext(joGetResult);
+                            }
+                            else {
+                                LOGINFO("%s is not available in the response", kDeviceInitializationContext);
+                            }
                         }
                         else {
-                            LOGINFO("%s is not available in the response", kDeviceInitializationContext);
+                            LOGINFO("getDeviceInitializationContext failed");
+                            if (!g_subscribed_for_deviceContextUpdate) {
+                                LOGINFO("onDeviceInitializationContextUpdate event not subscribed...");
+                                subscribeToDeviceInitializationEvent();
+                            }
                         }
                     }
                     else {
-                        LOGINFO("getDeviceInitializationContext failed");
+                        LOGINFO("Failed to get plugin handle");
                     }
+                    return success;
                 }
                 else {
-                    LOGINFO("Failed to get plugin handle");
+                    if (activation_status != "activated" && !g_subscribed_for_deviceContextUpdate) {
+                        LOGINFO("%s is not Active. Sleeping for %d seconds", secMgr_callsign, SECMANAGER_SLEEP_INTERVAL);
```

Comment:
LOGINFO("%s is not active. Retry after %d seconds", secMgr_callsign, SECMGR_RETRY_INTERVAL);